File: app/src/main/java/igrek/forceawaken/service/sensors/AccelerometerService.java
package igrek.forceawaken.service.sensors;

import android.app.Activity;
import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;

import javax.inject.Inject;

import igrek.forceawaken.dagger.DaggerIOC;
import igrek.forceawaken.logger.Logger;

public class AccelerometerService implements SensorEventListener {
	
	@Inject
	Activity activity;
	
	private float xValue, yValue, zValue;
	private Sensor accelerometer;
	
	@Inject
	Logger logger;
	
	public AccelerometerService() {
		DaggerIOC.getAppComponent().inject(this);
		logger.info("acxc stgatrted");
		SensorManager sensorManager = (SensorManager) activity.getSystemService(Context.SENSOR_SERVICE);
		if (sensorManager == null)
			logger.warn("no sensor manager found");
		else {
			accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
			if (accelerometer != null) {
				sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_NORMAL);
			}
		}
	}
	
	@Override
	public void onSensorChanged(SensorEvent event) {
		Sensor mySensor = event.sensor;
		if (mySensor.getType() == Sensor.TYPE_ACCELEROMETER) {
			xValue = event.values[0];
			yValue = event.values[1];
			zValue = event.values[2];
		}
	}
	
	@Override
	public void onAccuracyChanged(Sensor sensor, int accuracy) {
	
	}
	
	public Boolean isDeviceRotatedUp() {
		if (accelerometer == null)
			return null;
		return zValue > 0;
	}
}

File: app/src/main/java/igrek/forceawaken/sensors/AccelerometerService.kt
package igrek.forceawaken.sensors

import android.app.Activity
import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import igrek.forceawaken.inject.LazyExtractor
import igrek.forceawaken.inject.LazyInject
import igrek.forceawaken.inject.appFactory

class AccelerometerService(
        activity: LazyInject<Activity> = appFactory.activity,
) : SensorEventListener {
    private val activity by LazyExtractor(activity)

    private val sensorManager: SensorManager = activity.get().getSystemService(Context.SENSOR_SERVICE) as SensorManager

    private var xValue = 0f
    private var yValue = 0f
    private var zValue = 0f
    private var accelerometer: Sensor? = null
    override fun onSensorChanged(event: SensorEvent) {
        val mySensor: Sensor = event.sensor
        if (mySensor.type == Sensor.TYPE_ACCELEROMETER) {
            xValue = event.values.get(0)
            yValue = event.values.get(1)
            zValue = event.values.get(2)
        }
    }

    override fun onAccuracyChanged(sensor: Sensor, accuracy: Int) {}
    val isSpeakerRotatedDown: Boolean?
        get() = if (accelerometer == null) null else zValue > 0

    init {
        accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
        if (accelerometer != null) {
            sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_NORMAL)
        }
    }
}